BIC_codes/functions/dFC_funcs.py

```diff
@@ -0,0 +1,3472 @@
+    """
+    W_norm = W - np.min(W)
+    W_norm = np.divide(W_norm, np.max(W_norm))
+    return W_norm 
+
+# test
+def normalized_euc_dist(x, y):
+    # https://stats.stackexchange.com/questions/136232/definition-of-normalized-euclidean-distance#:~:text=The%20normalized%20squared%20euclidean%20distance,not%20related%20to%20Mahalanobis%20distance.
+
+    if np.linalg.norm(x-np.mean(x))**2==0 and np.linalg.norm(y-np.mean(y))**2==0:
+        return 0
+    return 0.5*((np.linalg.norm((x-np.mean(x)) - (y-np.mean(y)))**2)/(np.linalg.norm(x-np.mean(x))**2 + np.linalg.norm(y-np.mean(y))**2))
+
+def calc_ECM(A):
+    """
+    calc_ECM: Computes Eigenvector Centrality Mapping (ECM) 
+    of adjacency matrix A
+
+    Input:
+
+        A (np.array): adjacency matrix
+
+    Output:
+
+        centrality (np.array): ECM vector
+    """
+    G = nx.from_numpy_matrix(A) 
+    G.remove_edges_from(nx.selfloop_edges(G))
+    # G = G.to_undirected()
+    centrality = nx.eigenvector_centrality(G, weight='weight')
+    # centrality = nx.pagerank(G, alpha=0.85)
+
+    centrality = [centrality[node] for node in centrality]
+
+    return centrality
+
+# test
+def zip_name(name_lst):
+    # zip measure names
+    new_name_lst = list()
+    for name in name_lst:
+        if 'Clustering' in name:
+            new_name = 'SWC' + name[name.rfind('_'):]
+        if 'CAP' in name:
+            new_name = 'CAP' + name[name.rfind('_'):]
+        if 'ContinuousHMM' in name:
+            new_name = 'CHMM' + name[name.rfind('_'):]
+        if 'Windowless' in name:
+            new_name = 'WL' + name[name.rfind('_'):]
+        if 'DiscreteHMM' in name:
+            new_name = 'DHMM' + name[name.rfind('_'):]
+        if 'Time-Freq' in name:
+            new_name = 'TF' + name[name.rfind('_'):]
+        if 'SlidingWindow' in name:
+            new_name = 'SW' + name[name.rfind('_'):]
+        new_name_lst.append(new_name)
+    return new_name_lst
+
+# test
+# pear_corr problem
+def unzip_name(name):
+    # zip measure names
+    flag=False
+    if not '_' in name:
+        name = name + '_'
+        flag=True
+    if 'CAP' in name:
+        new_name = 'CAP' + name[name.rfind('_'):]
+    if 'SWC' in name:
+        new_name = 'Clustering' + name[name.rfind('_'):]
+    if 'CHMM' in name:
+        new_name = 'ContinuousHMM' + name[name.rfind('_'):]
+    if 'WL' in name:
+        new_name = 'Windowless' + name[name.rfind('_'):]
+    if 'DHMM' in name:
+        new_name = 'DiscreteHMM' + name[name.rfind('_'):]
+    if 'TF' in name:
+        new_name = 'Time-Freq' + name[name.rfind('_'):]
+    if 'SW_' in name:
+        new_name = 'SlidingWindow' + name[name.rfind('_'):]
+    if flag:
+        new_name = new_name[:-1]
+    return new_name
+
+#test
+def dFC_mat2vec(C_t):
+    '''
+    C_t must be an array of matrices or a single matrix
+    diagonal values not included. if you want to include 
+    them set k=0
+    '''
+    if len(C_t.shape)==2:
+        assert C_t.shape[0]==C_t.shape[1],\
```

Comment:
- maybe use a ValueError rather than assertion error
- break lines inside parentheses so you don't need `\`